client/my-sites/email/inbox/index.js

```diff
@@ -33,26 +36,48 @@ const InboxManagement = ( { domains } ) => {
 		);
 	}
 
-	const domainsWithSubscriptions = domains.filter(
-		( domain ) => ! domain.isWPCOMDomain && hasPaidEmailWithUs( domain )
+	// Find out if the site has at least one mailbox in the site
+	const hasAnyMailboxes = domains.some(
+		( domain ) =>
+			( ! domain.isWPCOMDomain &&
```

Comment:
Seeing as we are filtering against `domain.isWPCOMDomain` more than once in following scenarios, it may be worth it to have a pre-filtered set of domains i.e.

```js
const nonStagingDomains = domains.filter( domain => ! domain.isWPCOMDomain );
```

This way we won't have to do it twice, and the logic becomes easier to follow.